test/tap/tests/unit/genai_mysql_catalog_unit-t.cpp

```
@@ -0,0 +1,754 @@
+
+// ============================================================
+// 2. Upsert and Get
+// ============================================================
+
+static void test_upsert_and_get() {
+	MySQL_Catalog* cat = create_test_catalog();
+	ok(cat != nullptr, "catalog created for upsert_and_get");
+
+	// Insert an entry
+	int rc = cat->upsert("sales", "table", "orders",
+		"{\"summary\":\"Orders table with customer purchases\"}",
```

Cppcheck (2.20.0)

[warning] If memory allocation fails, then there is a possible null pointer dereference

(nullPointerOutOfMemory)